nemo/collections/tts/torch/data.py

```diff
@@ -490,6 +512,44 @@ def __getitem__(self, index):
 
             pitch_length = torch.tensor(len(pitch)).long()
 
+        # Load voiced_mask if needed
+        voiced_mask = None
+        if Voiced_mask in self.sup_data_types_set:
+            voiced_mask_path = self.voiced_mask_folder / f"{rel_audio_path_as_text_id}.pt"
+
+            if voiced_mask_path.exists():
+                voiced_mask = torch.load(voiced_mask_path).float()
+            else:
+                _, voiced_mask, _ = librosa.pyin(
+                    audio.numpy(),
+                    fmin=self.pitch_fmin,
+                    fmax=self.pitch_fmax,
+                    frame_length=self.win_length,
+                    sr=self.sample_rate,
+                    fill_na=0.0,
+                )
+                voiced_mask = torch.from_numpy(voiced_mask).float()
+                torch.save(voiced_mask, voiced_mask_path)
+
+        # Load p_voiced if needed
+        p_voiced = None
+        if P_voiced in self.sup_data_types_set:
+            p_voiced_path = self.p_voiced_folder / f"{rel_audio_path_as_text_id}.pt"
+
+            if p_voiced_path.exists():
+                p_voiced = torch.load(p_voiced_path).float()
+            else:
+                _, _, p_voiced = librosa.pyin(
+                    audio.numpy(),
+                    fmin=self.pitch_fmin,
+                    fmax=self.pitch_fmax,
+                    frame_length=self.win_length,
+                    sr=self.sample_rate,
+                    fill_na=0.0,
+                )
+                p_voiced = torch.from_numpy(p_voiced).float()
+                torch.save(p_voiced, p_voiced_path)
+
```

Comment:
the operations for `voiced_mask` and `p_voiced` are exactly the same. Could you please define a single member function and make a function call here to remove duplicates?

Reply:
Done.

Reply:
thank you.